### ISSUE TITLE
`on_validation_end` is not called, resulting in non saving models at each epoch

### ISSUE THREAD

**Issue by PonteIneptique** [mittagessen/kraken#343]
Right now, Kraken is supposed to save by overriding the `on_evaluation_end` of the pl.Trainer:

https://github.com/mittagessen/kraken/blob/0f82bbb9717eb2e5b9ffb6db6e842a1c1008c7f7/kraken/lib/train.py#L91-L104

As 1.6.0, this has been deprecated and will be removed in 1.8.0 of lightning (Cf. [source code of 1.6.0 of lightning](https://github.com/PyTorchLightning/pytorch-lightning/blob/60fe1520fd8ac8fa9006237f31045465200c9803/pytorch_lightning/trainer/callback_hook.py#L500-L511)).

On our training server, this straight resulted in *non saving models at all*

I have written a quick fix, available there: https://github.com/PonteIneptique/kraken/commit/572db5b855a4a550e1ad3d3b3d0d30fb19abfb38

I have not open a PR as this specific commits builds already on another PR, #342 

Pip freeze, python 3.7 (for kraken, latest master + my PR + a fix for this issue)
```
absl-py==1.0.0
aiohttp==3.8.1
aiosignal==1.2.0
albumentations==1.1.0
async-timeout==4.0.2
asynctest==0.13.0
attrs==21.4.0
cachetools==5.0.0
certifi==2021.10.8
charset-normalizer==2.0.12
click==8.1.2
commonmark==0.9.1
coremltools==5.2.0
frozenlist==1.3.0
fsspec==2022.3.0
google-auth==2.6.2
google-auth-oauthlib==0.4.6
grpcio==1.44.0
idna==3.3
imageio==2.16.1
importlib-metadata==4.11.3
importlib-resources==5.6.0
Jinja2==3.1.1
joblib==1.1.0
jsonschema==4.4.0
-e git+https://github.com/PonteIneptique/kraken.git@572db5b855a4a550e1ad3d3b3d0d30fb19abfb38#egg=kraken
lxml==4.8.0
Markdown==3.3.6
MarkupSafe==2.1.1
mpmath==1.2.1
multidict==6.0.2
networkx==2.6.3
numpy==1.21.5
oauthlib==3.2.0
opencv-python-headless==4.5.5.64
packaging==21.3
Pillow==9.1.0
protobuf==3.20.0
pyarrow==7.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pyDeprecate==0.3.2
Pygments==2.11.2
pyparsing==3.0.7
pyrsistent==0.18.1
python-bidi==0.4.2
pytorch-lightning==1.6.0
PyWavelets==1.3.0
PyYAML==6.0
qudida==0.0.4
regex==2022.3.15
requests==2.27.1
requests-oauthlib==1.3.1
rich==12.0.1
rsa==4.8
scikit-image==0.19.1
scikit-learn==1.0.2
scipy==1.7.3
Shapely==1.8.1.post1
six==1.16.0
sympy==1.10.1
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
threadpoolctl==3.1.0
tifffile==2021.11.2
torch==1.10.2
torchmetrics==0.7.3
torchvision==0.11.3
tqdm==4.63.1
typing_extensions==4.1.1
urllib3==1.26.9
Werkzeug==2.1.1
yarl==1.7.2
zipp==3.7.0
```

**Comment by mittagessen** [comment]
Hmm, this is weird. It works for me with pytorch-lightning 1.6.0 but yeah factoring it out into a callback is cleaner.

**Comment by PonteIneptique** [comment]
Crap. Then I don't know. I launch a dozen training trying to figure out the bug (three "normal" training, using 3 binaries files, the first one crashed in the end being not able to copy the best model as it was never written; the 7 or so others on a single small binary file to see what was going on. No weird arguments.), including by raising an exception in the Trainer method of `on_evaluation_end`, which was clearly never called as the exception did not break anything (I tried print, log, nothing).

Might be a combination of different factor (libraries or 3.7 ?) but that would be weird. I tested only on the 3.7 python server, I did not try to see if it bugged on my local computer.
